File: 1. Beginner/Golang/bee-1021.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {
	notes := [12]float64{100.00, 50.00, 20.00, 10.00, 5.00, 2.00, 1.00, 0.50, 0.25, 0.10, 0.05, 0.01}
	val := [12]int{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}

	var remains float64
	fmt.Scan(&remains)

	fmt.Println("NOTAS:")
	for i := 0; i < 6; i++ {
		if remains >= notes[i] {
			val[i] = int(remains / notes[i])
			remains, _ = strconv.ParseFloat(fmt.Sprintf("%.2f", math.Mod(remains, notes[i])), 64)
		}
		fmt.Printf("%d nota(s) de R$ %.2f\n", val[i], notes[i])
	}
	fmt.Println("MOEDAS:")
	for i := 6; i < 12; i++ {
		if remains >= notes[i] {
			val[i] = int(remains / notes[i])
			remains, _ = strconv.ParseFloat(fmt.Sprintf("%.2f", math.Mod(remains, notes[i])), 64)
		}
		fmt.Printf("%d moeda(s) de R$ %.2f\n", val[i], notes[i])
	}
}

File: 1. Beginner/Python/bee-1021.py
notes = [100.00, 50.00, 20.00, 10.00, 5.00, 2.00, 1.00, 0.50, 0.25, 0.10, 0.05, 0.01]
val =[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

remains = float(input())

def loop(start, end, str):

	global remains
	for i in range(start, end): 
		if remains >= notes[i]:
			val[i] = int(remains / notes[i])
			remains = round(remains % notes[i], 2)
		
		print(f"{val[i]} {str}(s) de R$ {notes[i]:.2f}")

print("NOTAS:")
loop(0,6, "nota")

print("MOEDAS:")
loop(6,12, "moeda")

